Add elevation layer to layerpicker

arottersman opened the pull request:
## Overview

* Add the painted elevation layer to the app
* 6ee745b shouldn't change any terrain results, but I will be deleting the old layer when this is on staging
* 6933093 fixes https://rollbar.com/Azavea/MMW/items/277/

Connects #2629 

### Demo

<img width="294" alt="screen shot 2018-03-02 at 11 01 22 am" src="https://user-images.githubusercontent.com/7633670/36911771-977881ae-1e12-11e8-8fb3-1ab510b08fac.png">

![screen shot 2018-03-01 at 2 51 11 pm](https://user-images.githubusercontent.com/7633670/36911774-97990fdc-1e12-11e8-94d8-a7a2fafb9c90.png)

![screen shot 2018-03-01 at 2 50 10 pm](https://user-images.githubusercontent.com/7633670/36911776-97ae5efa-1e12-11e8-89ec-161cdd82ab8a.png)
![screen shot 2018-03-01 at 2 49 52 pm](https://user-images.githubusercontent.com/7633670/36911777-97b87eb2-1e12-11e8-98ff-b373c777cb54.png)

## Testing Instructions

 * Pull and `bundle`
 * Restart celery, if necessary
 * Open the coverage grids in the layer selector and turn on "Elevation". Behold its beauty
 * Behold its layer legend
 * With the layer on, draw a shape where the layer does not exist (eg lake michigan or the florida keys). Confirm the terrain tab does not display an error
 * Draw a shape where the layer has some values and confirm the results match the painted version (PA should be <20m is, Rockies should be >2000m)

arottersman commented:
The build for this will fail until #2694 is in

rajadain commented:
@azavea-bot rebuild

arottersman commented:
I rebased on develop for fun.

rajadain commented:
Restarted Celery, ran Analyze around the country but I'm always getting 0 values for all Elevation:

![image](https://user-images.githubusercontent.com/1430060/36924598-42c0e270-1e3d-11e8-8752-aba28b485752.png)

Also, consider adding the Elevation layer to the Related Layers list as a dropdown, similar to the Water Quality tab:

![image](https://user-images.githubusercontent.com/1430060/36924626-59d9100e-1e3d-11e8-8968-327bd9b13ae6.png)

rajadain commented on src/mmw/apps/geoprocessing_api/tasks.py:
```diff
@@ -278,15 +278,18 @@ def analyze_terrain(result):
     [elevation, slope] = result
 
+    def cm_to_m(x):
+        x * M_PER_CM if x else None
```
Missing `return` is making all the elevation values 0.

arottersman replied:
Haha, that's embarrassing. Functional summer has done its work.

rajadain commented on src/mmw/mmw/settings/layer_settings.py:
```diff
@@ -148,6 +148,22 @@
+            'use_color_ramp': True,
+            'color_ramp_id': 'elevation-legend',
+            'legend_units_label': 'Elevation (m)',
+            'legend_unit_breaks': [-86, 0, 20, 100, 250, 750, 2000, 4413],
```
This is a stylistic thing, but I think the legend will look better with more regular numbers. Since the numbers are fuzzy on the gradient line, can we make the lowest number `-100` and the highest number `4500`? We can also go with `-80` and `4400` for a compromise between accuracy and readability.

arottersman replied:
I have no preference on this, any thoughts, @ajrobbins? I think if we round the min/max here, we  should also round them on slope percent
<img width="382" alt="screen shot 2018-03-05 at 10 21 54 am" src="https://user-images.githubusercontent.com/7633670/36982989-3bf714b8-205f-11e8-8407-614958a61992.png">

arottersman replied:
Arianna and I chatted in the kitchen, and we like being able to see the min and max on the scale; going to keep this is as for now.

arottersman commented:
* Fixed the missing `return`
* Added to related layers
<img width="254" alt="screen shot 2018-03-05 at 11 20 37 am" src="https://user-images.githubusercontent.com/7633670/36986228-5df425c6-2067-11e8-8264-d6242b6e7f08.png">
* Reordered `Elevation` and `Slope` in layerpicker (Elevation felt like it should come first because "E" < "S" and slope is derived from the elevation data

rajadain commented:
Taking another look.

rajadain commented on src/mmw/js/src/analyze/views.js:
```diff
@@ -1314,7 +1314,7 @@ var TerrainResultView = AnalyzeResultView.extend({
             source = 'NHDPlus V2 NEDSnapshot DEM',
             helpText = 'For more information and data sources, see <a href=\'https://wikiwatershed.org/documentation/mmw-tech/#overlays-tab-coverage\' target=\'_blank\' rel=\'noreferrer noopener\'>Model My Watershed Technical Documentation on Coverage Grids</a>',
-            associatedLayerCodes = ['pct_slope'],
+            associatedLayerCodes = ['elevation', 'pct_slope'],
```
Quite brilliant that just by making it a list of more than one the UI auto-adjusts

rajadain commented:
Friendly reminder to squash the fixups before merging.

arottersman commented:
Thanks for the helpful review, Terence!

arottersman commented:
Going to delete the old layer after RRP

rajadain commented:
Was the old layer deleted?

arottersman commented:
Nope! It's time.

arottersman commented:
Okay, deleted. Thanks for reminding me!
